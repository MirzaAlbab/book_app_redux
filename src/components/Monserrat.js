import React from 'react';
import {Text} from 'react-native';
// import {ms} from 'react-native-size-matters';

const Monserrat = ({
  type = 'Regular',
  children,
  color = 'black',
  size = 12,
  ...style
}) => {
  return (
    <Text
      style={{
        fontFamily: `Montserrat-${type}`,
        color: color,
        fontSize: size,
        ...style,
      }}>
      {children}
    </Text>
  );
};
export default Monserrat;
