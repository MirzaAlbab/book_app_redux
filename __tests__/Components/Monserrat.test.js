import React from 'react';
import Monserrat from '../../src/components/Monserrat';
import {create} from 'react-test-renderer';

describe('Monserrat Testing', () => {
  describe('Monserrat', () => {
    it('renders correctly', () => {
      const tree = create(<Monserrat />);
      expect(tree).toMatchSnapshot();
    });
    it('renders correctly with props', () => {
      const tree = create(<Monserrat type="Bold" color="red" size={20} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
