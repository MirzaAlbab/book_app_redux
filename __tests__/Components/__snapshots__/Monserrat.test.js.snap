// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monserrat Testing Monserrat renders correctly 1`] = `
<Text
  style={
    Object {
      "color": "black",
      "fontFamily": "Montserrat-Regular",
      "fontSize": 12,
    }
  }
/>
`;

exports[`Monserrat Testing Monserrat renders correctly with props 1`] = `
<Text
  style={
    Object {
      "color": "red",
      "fontFamily": "Montserrat-Bold",
      "fontSize": 20,
    }
  }
/>
`;
